sdk/core/azure-core-test/src/main/java/com/azure/core/test/http/TestProxyPlaybackClient.java

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.http;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.test.models.TestProxyMatcher;
+import com.azure.core.test.models.TestProxySanitizer;
+import com.azure.core.test.utils.HttpURLConnectionHttpClient;
+import com.azure.core.test.utils.TestProxyUtils;
+import com.azure.core.util.Context;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.stream.Collectors;
+
+import static com.azure.core.test.utils.TestProxyUtils.getMatcherRequests;
+import static com.azure.core.test.utils.TestProxyUtils.getSanitizerRequests;
+import static com.azure.core.test.utils.TestProxyUtils.loadMatchers;
+import static com.azure.core.test.utils.TestProxyUtils.loadSanitizers;
+
+/**
+ * A {@link HttpClient} that plays back test recordings from the external test proxy.
+ */
+public class TestProxyPlaybackClient implements HttpClient {
+
+    private final HttpURLConnectionHttpClient client = new HttpURLConnectionHttpClient();
```

Comment:
Long-term is there plans on making this pluggable?

Reply:
Much shorter term in fact. :)